pkg/registry/batch/cronjob/storage/storage_test.go

```diff
@@ -87,7 +82,18 @@ func TestCreate(t *testing.T) {
 		validCronJob,
 		// invalid (empty spec)
 		&batch.CronJob{
-			Spec: batch.CronJobSpec{},
+			Spec: batch.CronJobSpec{
+				// FIX ME
```

Comment:
If we don't, we panic instead. I think I can provide something cleaner using MakePod. Notice its not obvious what we expect to fail but a panic isn't it.

Reply:
Yeah, panic seems wrong.  Can you stick the backtrace here?

Reply:
Ahh, right.

Why don't we flag the error and pass the pointer (including when it is nil) around instead of the value.  The few places that need it can check for nil and simply skip their checks, since it has already been flagged.

We want validation to report as many errors as it can, but if this is nil, some checks are actually impossible.

```
      if spec.TerminationGracePeriodSeconds == nil {
          allErrs = append(allErrs, field.Required(fldPath.Child("terminationGracePeriodSeconds"), ""))
      }
```` 

Or alternately, we could flag the error and then pass around a default, but that will end up in weird cases where we compare against the default, producing errors that will be hard to understand by users.

Reply:
Adding the check is the easy part. The issue is all the compares we do later with gracePeriod against the lifecycle and probes, e.g. preStop, livenessProbe, etc...
I can re-plumb all that to skip the checks where we actually care.

Reply:
it seems to work.

Reply:
yeah, all of those can just check for nil and skip the check.  It gets passed around a lot but not many actual checks.  Actuall just one, if I am looking correctly.